lms/djangoapps/courseware/views/views.py

```diff
@@ -726,6 +726,12 @@ def _progress(request, course_key, student_id):
 
     # checking certificate generation configuration
     enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(student, course_key)
+
+    # We need to have id verified to show generate cert button
+    # in the case where courseMode is of Verified variety.
+    is_id_verified = SoftwareSecurePhotoVerification.user_is_verified(student) or \
+        enrollment_mode and not CourseMode.has_verified_mode(enrollment_mode)
```

Comment:
@adampalay The logic here is not excluding base on what course modes the course have, it's excluding base on what enrollment mode the student is in currently. So you comment is not really relevant to the conditions here. As your comment on the name of the property `is_id_verified`, that is a fair comment. I will update name.